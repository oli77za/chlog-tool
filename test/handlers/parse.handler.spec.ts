const readFileSyncMock = jest.fn().mockReturnValue('');
jest.mock('fs', () => ({
  readFileSync: readFileSyncMock,
}));

const parseFunctionMock = jest.fn().mockReturnValue({ foo: 'bar' });
jest.mock('../../src/parser', () => parseFunctionMock);

const writeSpy = jest
  .spyOn(process.stdout, 'write')
  .mockImplementation(() => true);

import parse from '../../src/handlers/parse.handler';

describe('Handlers / Parse', () => {
  beforeEach(() => jest.clearAllMocks());

  it('Should parse from stdin', () => {
    const parseArgs = {} as any;
    parse(parseArgs);
    expect(readFileSyncMock.mock.calls[0][0]).toBe(0);
    expect(writeSpy.mock.calls).toMatchSnapshot();
  });

  it('Should parse from input file', () => {
    const parseArgs = { input: 'testinputfile' } as any;
    parse(parseArgs);
    expect(readFileSyncMock.mock.calls[0][0]).toBe('testinputfile');
    expect(writeSpy.mock.calls).toMatchSnapshot();
  });

  it('Should pretty print the output', () => {
    const parseArgs = { pretty: true } as any;
    parse(parseArgs);
    expect(writeSpy.mock.calls).toMatchSnapshot();
  });
});
