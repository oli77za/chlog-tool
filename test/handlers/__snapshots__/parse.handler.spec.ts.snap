// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handlers / Parse Should parse from input file 1`] = `
[
  [
    "{"foo":"bar"}",
  ],
]
`;

exports[`Handlers / Parse Should parse from stdin 1`] = `
[
  [
    "{"foo":"bar"}",
  ],
]
`;

exports[`Handlers / Parse Should pretty print the output 1`] = `
[
  [
    "{
  "foo": "bar"
}",
  ],
]
`;
